### PR TITLE
Run goreleaser action in snapshot mode from merge queue

### PR DESCRIPTION
**Pull request by pietern**
## Changes

Running this action before doing an actual release should prevent situations like #3645 where it doesn't work on release.

## Tests

See workflows.

### CODE REVIEW

**Comment by pietern** [comment]
This change still includes the pull request trigger to test it out, but must be removed before merging.

We can't run the goreleaser action on PRs because PRs from forks don't have access to secrets.

**Comment by pietern** [comment]
It works: https://github.com/databricks/terraform-provider-databricks/actions/runs/9381922308/job/25832223352

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -19,21 +21,25 @@ jobs:
...
 
-      - name: Pull external libraries
-        run: make vendor
-
       - name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v6
         with:
           version: ~> v2
-          args: release --clean
+          args: release --clean ${{ !startsWith(github.ref, 'refs/tags/v') && '--snapshot' || '' }}
```
Comment (pietern) [inline .github/workflows/release.yml:42]: Also see https://docs.github.com/en/actions/learn-github-actions/expressions#example

Comment (mgyucht) [reply .github/workflows/release.yml:42]: What is `github.ref`'s value when something is in the merge queue?

**Review by mgyucht — APPROVED** [review]
Does the merge fail if this fails? It's not a required check. Still good that we at least run it once.

File: .github/workflows/release.yml (hunk 1 above)
Comment (pietern) [reply .github/workflows/release.yml:42]: Something like `gh-readonly-queue/main/...`, definitely not `refs/tags/...`.

**Pull request merged**